controllers/modals/submission/ViewSubmissionMetadataHandler.inc.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * @file controllers/modals/submission/ViewSubmissionMetadataHandler.inc.php
+ *
+ * Copyright (c) 2014-2019 Simon Fraser University
+ * Copyright (c) 2003-2019 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class ViewSubmissionMetadataHandler
+ * @ingroup controllers_modals_viewSubmissionMetadataHandler
+ *
+ * @brief Display submission metadata.
+ */
+
+// Import the base Handler.
+import('classes.handler.Handler');
+
+class ViewSubmissionMetadataHandler extends handler {
+
+	/**
+	 * Constructor.
+	 */
+	function __construct() {
+		parent::__construct();
+		$this->addRoleAssignment(array(ROLE_ID_REVIEWER), array('display'));
+	}
+
+	/**
+	 * @copydoc PKPHandler::authorize()
+	 */
+	function authorize($request, &$args, $roleAssignments) {
+		import('lib.pkp.classes.security.authorization.SubmissionAccessPolicy');
+		$this->addPolicy(new SubmissionAccessPolicy($request, $args, $roleAssignments));
+		return parent::authorize($request, $args, $roleAssignments);
+	}
+
+	/**
+	 * Display metadata
+	 */
+	function display($args, $request) {
+		$submission = $this->getAuthorizedContextObject(ASSOC_TYPE_SUBMISSION);
+		$reviewAssignment = $this->getAuthorizedContextObject(ASSOC_TYPE_REVIEW_ASSIGNMENT);
+		$templateMgr = TemplateManager::getManager($request);
+
+		if ($reviewAssignment->getReviewMethod() != SUBMISSION_REVIEW_METHOD_DOUBLEBLIND) { /* SUBMISSION_REVIEW_METHOD_BLIND or _OPEN */
+			$templateMgr->assign('authors', $submission->getAuthorString());
```

Comment:
Although `$submission->getAuthorString()` is still around (and used), I've deprecated it in favor of a method on the publication:

```
$publication->getAuthorString($userGroups);
```

Pass in the user groups for the context, eg - `DAORegistry::getDAO('UserGroupDAO')->getByContextId($contextId)`. This prevents the database lookup from occurring every time the author string is compiled. (Which is only once in your case but not in others.)

Reply:
so having
$userGroupDao = DAORegistry::getDAO('UserGroupDAO');
$userGroups = $userGroupDao->getByContextId($context->getId());
$templateMgr->assign('authors', $publication->getAuthorString($userGroups));

Gives
PHP Fatal error:  Uncaught Error: Call to undefined method UserGroupDAO::getId() in lib/pkp/classes/publication/PKPPublication.inc.php:162

Reply:
I have that in the current pr, but it gives the error above if the code is triggered.

Reply:
Sorry, I always forget the DAOs return a factory. This will give you the array of objects:

```
DAORegistry::getDAO('UserGroupDAO')->getByContextId($contextId)->toArray();
```

Reply:
check